java-checks/src/main/java/org/sonar/java/checks/maven/helpers/MavenDependencyVersionMatcher.java

```diff
@@ -0,0 +1,218 @@
+      return matchesLowerBound && matchesUpperBound;
+    }
+
+    /**
+     * Build a {@link ArtifactVersion} from a String, without failing in case of exception while parsing.
+     * @param version the raw version as string
+     * @return the {@link ArtifactVersion} corresponding to the provided version as string
+     */
+    @CheckForNull
+    private static ArtifactVersion getVersionSilently(String version) {
+      try {
+        return ArtifactVersion.parseString(version);
+      } catch (NumberFormatException e) {
+        // do nothing, version will be ignored
+      }
+      return null;
+    }
+
+    /**
+     * Build a {@link ArtifactVersion} from a String, throwing an {@link IllegalArgumentException} if failed to parse value.
+     * @param version the raw version as string
+     * @param callingCheckKey the check requiring the matcher
+     * @return the {@link ArtifactVersion} corresponding to the provided version as string
+     */
+    private static ArtifactVersion getVersion(String version, String callingCheckKey) {
+      try {
+        return ArtifactVersion.parseString(version);
+      } catch (NumberFormatException e) {
+        throw new IllegalArgumentException("[" + callingCheckKey + "] Provided version does not match expected pattern:"
+          + " <major version>.<minor version>.<incremental version> (recieved: " + version + ")", e);
+      }
+    }
+
+    /**
+     * Expected format: <code>[major version].[minor version].[incremental version]-[qualifier]</code>
+     *
+     * <br />
+     *
+     * Handled formats:
+     * <ul>
+     *   <li><code>1.0.1-SNAPSHOT</code></li>
+     *   <li><code>2.4</code></li>
+     *   <li><code>1</code></li>
+     *   <li><code>0.5-alpha</code></li>
+     * </ul>
+     */
+    private static class ArtifactVersion {
+      private static final int HANDLED_VERSIONS = 3;
+      private final Integer[] versions = new Integer[HANDLED_VERSIONS];
+
+      private static ArtifactVersion parseString(String version) {
+        String[] split = version.split("\\.");
+        if (version.contains("-")) {
+          // ignore qualifier
+          split = version.split("-")[0].split("\\.");
+        }
+        ArtifactVersion result = new ArtifactVersion();
+        for (int i = 0; i < Math.min(HANDLED_VERSIONS, split.length); i++) {
+          result.versions[i] = Integer.parseInt(split[i]);
+        }
+        return result;
+      }
+
+      public boolean isAfterOrEqual(ArtifactVersion v) {
+        return isAfterOrEqual(v, 0);
+      }
+
+      public boolean isAfterOrEqual(ArtifactVersion v, int i) {
+        if (i >= HANDLED_VERSIONS) {
+          return true;
+        }
+        Integer local = versions[i];
+        Integer target = v.versions[i];
+
+        if (target == null) {
+          return true;
+        } else if (local == null) {
+          return false;
+        }
+
+        return (local > target) || (local == target && isAfterOrEqual(v, i+1));
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this comparison to use the equals method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1698)